pkg/operators/kubeipresolver/kubeipresolver.go

```diff
@@ -105,57 +103,25 @@ func (m *KubeIPResolverInstance) PostGadgetRun() error {
 }
 
 func (m *KubeIPResolverInstance) enrich(ev any) {
-	pods, err := m.k8sInventory.GetPods()
-	if err != nil {
-		log.Warnf("getting pods from k8s inventory: %v", err)
-		return
-	}
-
 	endpoints := ev.(KubeIPResolverInterface).GetEndpoints()
-	for j := range endpoints {
-		// initialize to this default value if we don't find a match
-		endpoints[j].Kind = types.EndpointKindRaw
-	}
-
-	found := 0
-	for _, pod := range pods {
-		if pod.Spec.HostNetwork {
+	for _, endpoint := range endpoints {
+		endpoint.Kind = types.EndpointKindRaw
+
+		pod := m.k8sInventory.GetPodByIp(endpoint.Addr)
+		if pod != nil {
+			endpoint.Kind = types.EndpointKindPod
```

Comment:
Verify if the pod is using the [host network ns here](https://github.com/inspektor-gadget/inspektor-gadget/blob/f5d4a91c71fee252472d593dc96c4b4bff1d318e/pkg/operators/kubeipresolver/kubeipresolver.go#L122-L124)?